// @n8n/rest-api-client/dist/sso.cjs
const require_utils = require('./utils2.cjs');

//#region src/api/sso.ts
const initSSO = async (context, redirectUrl = "") => {
	return await require_utils.makeRestApiRequest(context, "GET", `/sso/saml/initsso?redirect=${redirectUrl}`);
};
const getSamlMetadata = async (context) => {
	return await require_utils.makeRestApiRequest(context, "GET", "/sso/saml/metadata");
};
const getSamlConfig = async (context) => {
	return await require_utils.makeRestApiRequest(context, "GET", "/sso/saml/config");
};
const saveSamlConfig = async (context, data) => {
	return await require_utils.makeRestApiRequest(context, "POST", "/sso/saml/config", data);
};
const toggleSamlConfig = async (context, data) => {
	return await require_utils.makeRestApiRequest(context, "POST", "/sso/saml/config/toggle", data);
};
const testSamlConfig = async (context) => {
	return await require_utils.makeRestApiRequest(context, "GET", "/sso/saml/config/test");
};
const getOidcConfig = async (context) => {
	return await require_utils.makeRestApiRequest(context, "GET", "/sso/oidc/config");
};
const saveOidcConfig = async (context, data) => {
	return await require_utils.makeRestApiRequest(context, "POST", "/sso/oidc/config", data);
};
const initOidcLogin = async (context) => {
	return await require_utils.makeRestApiRequest(context, "GET", "/sso/oidc/login");
};

//#endregion
Object.defineProperty(exports, 'getOidcConfig', {
  enumerable: true,
  get: function () {
    return getOidcConfig;
  }
});
Object.defineProperty(exports, 'getSamlConfig', {
  enumerable: true,
  get: function () {
    return getSamlConfig;
  }
});
Object.defineProperty(exports, 'getSamlMetadata', {
  enumerable: true,
  get: function () {
    return getSamlMetadata;
  }
});
Object.defineProperty(exports, 'initOidcLogin', {
  enumerable: true,
  get: function () {
    return initOidcLogin;
  }
});
Object.defineProperty(exports, 'initSSO', {
  enumerable: true,
  get: function () {
    return initSSO;
  }
});
Object.defineProperty(exports, 'saveOidcConfig', {
  enumerable: true,
  get: function () {
    return saveOidcConfig;
  }
});
Object.defineProperty(exports, 'saveSamlConfig', {
  enumerable: true,
  get: function () {
    return saveSamlConfig;
  }
});
Object.defineProperty(exports, 'testSamlConfig', {
  enumerable: true,
  get: function () {
    return testSamlConfig;
  }
});
Object.defineProperty(exports, 'toggleSamlConfig', {
  enumerable: true,
  get: function () {
    return toggleSamlConfig;
  }
});
