// @n8n/rest-api-client/dist/users.cjs
const require_utils = require('./utils2.cjs');

//#region src/api/users.ts
async function loginCurrentUser(context) {
	return await require_utils.makeRestApiRequest(context, "GET", "/login");
}
async function login(context, params) {
	return await require_utils.makeRestApiRequest(context, "POST", "/login", params);
}
async function logout(context) {
	await require_utils.makeRestApiRequest(context, "POST", "/logout");
}
async function setupOwner(context, params) {
	return await require_utils.makeRestApiRequest(context, "POST", "/owner/setup", params);
}
async function validateSignupToken(context, params) {
	return await require_utils.makeRestApiRequest(context, "GET", "/resolve-signup-token", params);
}
async function signup(context, params) {
	const { inviteeId, ...props } = params;
	return await require_utils.makeRestApiRequest(context, "POST", `/users/${params.inviteeId}`, props);
}
async function sendForgotPasswordEmail(context, params) {
	await require_utils.makeRestApiRequest(context, "POST", "/forgot-password", params);
}
async function validatePasswordToken(context, params) {
	await require_utils.makeRestApiRequest(context, "GET", "/resolve-password-token", params);
}
async function changePassword(context, params) {
	await require_utils.makeRestApiRequest(context, "POST", "/change-password", params);
}
async function updateCurrentUser(context, params) {
	return await require_utils.makeRestApiRequest(context, "PATCH", "/me", params);
}
async function updateCurrentUserSettings(context, settings) {
	return await require_utils.makeRestApiRequest(context, "PATCH", "/me/settings", settings);
}
async function updateOtherUserSettings(context, userId, settings) {
	return await require_utils.makeRestApiRequest(context, "PATCH", `/users/${userId}/settings`, settings);
}
async function updateCurrentUserPassword(context, params) {
	return await require_utils.makeRestApiRequest(context, "PATCH", "/me/password", params);
}
async function deleteUser(context, { id, transferId }) {
	await require_utils.makeRestApiRequest(context, "DELETE", `/users/${id}`, transferId ? { transferId } : {});
}
async function getUsers(context, filter) {
	return await require_utils.makeRestApiRequest(context, "GET", "/users", filter);
}
async function getInviteLink(context, { id }) {
	return await require_utils.makeRestApiRequest(context, "GET", `/users/${id}/invite-link`);
}
async function generateInviteLink(context, { id }) {
	return await require_utils.makeRestApiRequest(context, "POST", `/users/${id}/invite-link`);
}
async function getPasswordResetLink(context, { id }) {
	return await require_utils.makeRestApiRequest(context, "GET", `/users/${id}/password-reset-link`);
}
async function submitPersonalizationSurvey(context, params) {
	await require_utils.makeRestApiRequest(context, "POST", "/me/survey", params);
}
async function updateGlobalRole(context, { id, newRoleName }) {
	return await require_utils.makeRestApiRequest(context, "PATCH", `/users/${id}/role`, { newRoleName });
}

//#endregion
Object.defineProperty(exports, 'changePassword', {
  enumerable: true,
  get: function () {
    return changePassword;
  }
});
Object.defineProperty(exports, 'deleteUser', {
  enumerable: true,
  get: function () {
    return deleteUser;
  }
});
Object.defineProperty(exports, 'generateInviteLink', {
  enumerable: true,
  get: function () {
    return generateInviteLink;
  }
});
Object.defineProperty(exports, 'getInviteLink', {
  enumerable: true,
  get: function () {
    return getInviteLink;
  }
});
Object.defineProperty(exports, 'getPasswordResetLink', {
  enumerable: true,
  get: function () {
    return getPasswordResetLink;
  }
});
Object.defineProperty(exports, 'getUsers', {
  enumerable: true,
  get: function () {
    return getUsers;
  }
});
Object.defineProperty(exports, 'login', {
  enumerable: true,
  get: function () {
    return login;
  }
});
Object.defineProperty(exports, 'loginCurrentUser', {
  enumerable: true,
  get: function () {
    return loginCurrentUser;
  }
});
Object.defineProperty(exports, 'logout', {
  enumerable: true,
  get: function () {
    return logout;
  }
});
Object.defineProperty(exports, 'sendForgotPasswordEmail', {
  enumerable: true,
  get: function () {
    return sendForgotPasswordEmail;
  }
});
Object.defineProperty(exports, 'setupOwner', {
  enumerable: true,
  get: function () {
    return setupOwner;
  }
});
Object.defineProperty(exports, 'signup', {
  enumerable: true,
  get: function () {
    return signup;
  }
});
Object.defineProperty(exports, 'submitPersonalizationSurvey', {
  enumerable: true,
  get: function () {
    return submitPersonalizationSurvey;
  }
});
Object.defineProperty(exports, 'updateCurrentUser', {
  enumerable: true,
  get: function () {
    return updateCurrentUser;
  }
});
Object.defineProperty(exports, 'updateCurrentUserPassword', {
  enumerable: true,
  get: function () {
    return updateCurrentUserPassword;
  }
});
Object.defineProperty(exports, 'updateCurrentUserSettings', {
  enumerable: true,
  get: function () {
    return updateCurrentUserSettings;
  }
});
Object.defineProperty(exports, 'updateGlobalRole', {
  enumerable: true,
  get: function () {
    return updateGlobalRole;
  }
});
Object.defineProperty(exports, 'updateOtherUserSettings', {
  enumerable: true,
  get: function () {
    return updateOtherUserSettings;
  }
});
Object.defineProperty(exports, 'validatePasswordToken', {
  enumerable: true,
  get: function () {
    return validatePasswordToken;
  }
});
Object.defineProperty(exports, 'validateSignupToken', {
  enumerable: true,
  get: function () {
    return validateSignupToken;
  }
});
