// @n8n/rest-api-client/dist/api/index.cjs
require('../utils2.cjs');
const require_api_keys = require('../api-keys.cjs');
const require_cloudPlans = require('../cloudPlans.cjs');
const require_communityNodes = require('../communityNodes.cjs');
const require_credentialResolvers = require('../credentialResolvers.cjs');
const require_ctas = require('../ctas.cjs');
const require_eventbus_ee = require('../eventbus.ee.cjs');
const require_events = require('../events.cjs');
const require_externalSecrets_ee = require('../externalSecrets.ee.cjs');
const require_ldap = require('../ldap.cjs');
const require_mfa = require('../mfa.cjs');
const require_nodeTypes = require('../nodeTypes.cjs');
const require_npsSurvey = require('../npsSurvey.cjs');
const require_orchestration = require('../orchestration.cjs');
const require_provisioning = require('../provisioning.cjs');
const require_prompts = require('../prompts.cjs');
const require_roles = require('../roles.cjs');
const require_settings = require('../settings.cjs');
const require_module_settings = require('../module-settings.cjs');
const require_sso = require('../sso.cjs');
const require_templates = require('../templates.cjs');
const require_third_party_licenses = require('../third-party-licenses.cjs');
const require_ui = require('../ui.cjs');
const require_usage = require('../usage.cjs');
const require_users = require('../users.cjs');
const require_versions = require('../versions.cjs');
const require_webhooks = require('../webhooks.cjs');
const require_workflowHistory = require('../workflowHistory.cjs');
require('../api.cjs');

exports.activateLicenseKey = require_usage.activateLicenseKey;
exports.canEnableMFA = require_mfa.canEnableMFA;
exports.changePassword = require_users.changePassword;
exports.connectProvider = require_externalSecrets_ee.connectProvider;
exports.createApiKey = require_api_keys.createApiKey;
exports.createCredentialResolver = require_credentialResolvers.createCredentialResolver;
exports.createProjectRole = require_roles.createProjectRole;
exports.deleteApiKey = require_api_keys.deleteApiKey;
exports.deleteCredentialResolver = require_credentialResolvers.deleteCredentialResolver;
exports.deleteDestinationFromDb = require_eventbus_ee.deleteDestinationFromDb;
exports.deleteProjectRole = require_roles.deleteProjectRole;
exports.deleteUser = require_users.deleteUser;
exports.disableMfa = require_mfa.disableMfa;
exports.dismissBannerPermanently = require_ui.dismissBannerPermanently;
exports.enableMfa = require_mfa.enableMfa;
exports.fetchCommunityNodeAttributes = require_nodeTypes.fetchCommunityNodeAttributes;
exports.fetchCommunityNodeTypes = require_nodeTypes.fetchCommunityNodeTypes;
exports.findWebhook = require_webhooks.findWebhook;
exports.generateInviteLink = require_users.generateInviteLink;
exports.getAdminPanelLoginCode = require_cloudPlans.getAdminPanelLoginCode;
exports.getApiKeyScopes = require_api_keys.getApiKeyScopes;
exports.getApiKeys = require_api_keys.getApiKeys;
exports.getAvailableCommunityPackageCount = require_communityNodes.getAvailableCommunityPackageCount;
exports.getBecomeCreatorCta = require_ctas.getBecomeCreatorCta;
exports.getCategories = require_templates.getCategories;
exports.getCloudUserInfo = require_cloudPlans.getCloudUserInfo;
exports.getCollectionById = require_templates.getCollectionById;
exports.getCollections = require_templates.getCollections;
exports.getCredentialResolver = require_credentialResolvers.getCredentialResolver;
exports.getCredentialResolverTypes = require_credentialResolvers.getCredentialResolverTypes;
exports.getCredentialResolvers = require_credentialResolvers.getCredentialResolvers;
exports.getCurrentPlan = require_cloudPlans.getCurrentPlan;
exports.getCurrentUsage = require_cloudPlans.getCurrentUsage;
exports.getDestinationsFromBackend = require_eventbus_ee.getDestinationsFromBackend;
exports.getEventNamesFromBackend = require_eventbus_ee.getEventNamesFromBackend;
exports.getExternalSecrets = require_externalSecrets_ee.getExternalSecrets;
exports.getExternalSecretsProvider = require_externalSecrets_ee.getExternalSecretsProvider;
exports.getExternalSecretsProviders = require_externalSecrets_ee.getExternalSecretsProviders;
exports.getInstalledCommunityNodes = require_communityNodes.getInstalledCommunityNodes;
exports.getInviteLink = require_users.getInviteLink;
exports.getLdapConfig = require_ldap.getLdapConfig;
exports.getLdapSynchronizations = require_ldap.getLdapSynchronizations;
exports.getLicense = require_usage.getLicense;
exports.getLocalResourceMapperFields = require_nodeTypes.getLocalResourceMapperFields;
exports.getMfaQR = require_mfa.getMfaQR;
exports.getModuleSettings = require_module_settings.getModuleSettings;
exports.getNextVersions = require_versions.getNextVersions;
exports.getNodeParameterActionResult = require_nodeTypes.getNodeParameterActionResult;
exports.getNodeParameterOptions = require_nodeTypes.getNodeParameterOptions;
exports.getNodeTranslationHeaders = require_nodeTypes.getNodeTranslationHeaders;
exports.getNodeTypeVersions = require_nodeTypes.getNodeTypeVersions;
exports.getNodeTypes = require_nodeTypes.getNodeTypes;
exports.getNodeTypesByIdentifier = require_nodeTypes.getNodeTypesByIdentifier;
exports.getNodesInformation = require_nodeTypes.getNodesInformation;
exports.getOidcConfig = require_sso.getOidcConfig;
exports.getPasswordResetLink = require_users.getPasswordResetLink;
exports.getPromptsData = require_prompts.getPromptsData;
exports.getProvisioningConfig = require_provisioning.getProvisioningConfig;
exports.getResourceLocatorResults = require_nodeTypes.getResourceLocatorResults;
exports.getResourceMapperFields = require_nodeTypes.getResourceMapperFields;
exports.getRoleBySlug = require_roles.getRoleBySlug;
exports.getRoles = require_roles.getRoles;
exports.getSamlConfig = require_sso.getSamlConfig;
exports.getSamlMetadata = require_sso.getSamlMetadata;
exports.getSettings = require_settings.getSettings;
exports.getTemplateById = require_templates.getTemplateById;
exports.getThirdPartyLicenses = require_third_party_licenses.getThirdPartyLicenses;
exports.getUsers = require_users.getUsers;
exports.getWhatsNewSection = require_versions.getWhatsNewSection;
exports.getWorkflowHistory = require_workflowHistory.getWorkflowHistory;
exports.getWorkflowTemplate = require_templates.getWorkflowTemplate;
exports.getWorkflowVersion = require_workflowHistory.getWorkflowVersion;
exports.getWorkflowVersionsByIds = require_workflowHistory.getWorkflowVersionsByIds;
exports.getWorkflows = require_templates.getWorkflows;
exports.hasDestinationId = require_eventbus_ee.hasDestinationId;
exports.initOidcLogin = require_sso.initOidcLogin;
exports.initSSO = require_sso.initSSO;
exports.installNewPackage = require_communityNodes.installNewPackage;
exports.login = require_users.login;
exports.loginCurrentUser = require_users.loginCurrentUser;
exports.logout = require_users.logout;
exports.registerCommunityEdition = require_usage.registerCommunityEdition;
exports.reloadProvider = require_externalSecrets_ee.reloadProvider;
exports.renewLicense = require_usage.renewLicense;
exports.requestLicenseTrial = require_usage.requestLicenseTrial;
exports.runLdapSync = require_ldap.runLdapSync;
exports.saveDestinationToDb = require_eventbus_ee.saveDestinationToDb;
exports.saveOidcConfig = require_sso.saveOidcConfig;
exports.saveProvisioningConfig = require_provisioning.saveProvisioningConfig;
exports.saveSamlConfig = require_sso.saveSamlConfig;
exports.sendConfirmationEmail = require_cloudPlans.sendConfirmationEmail;
exports.sendForgotPasswordEmail = require_users.sendForgotPasswordEmail;
exports.sendGetWorkerStatus = require_orchestration.sendGetWorkerStatus;
exports.sendTestMessageToDestination = require_eventbus_ee.sendTestMessageToDestination;
exports.sendUserEvent = require_cloudPlans.sendUserEvent;
exports.sessionStarted = require_events.sessionStarted;
exports.setupOwner = require_users.setupOwner;
exports.signup = require_users.signup;
exports.submitContactInfo = require_prompts.submitContactInfo;
exports.submitPersonalizationSurvey = require_users.submitPersonalizationSurvey;
exports.testExternalSecretsProviderConnection = require_externalSecrets_ee.testExternalSecretsProviderConnection;
exports.testHealthEndpoint = require_templates.testHealthEndpoint;
exports.testLdapConnection = require_ldap.testLdapConnection;
exports.testSamlConfig = require_sso.testSamlConfig;
exports.toggleSamlConfig = require_sso.toggleSamlConfig;
exports.uninstallPackage = require_communityNodes.uninstallPackage;
exports.updateApiKey = require_api_keys.updateApiKey;
exports.updateCredentialResolver = require_credentialResolvers.updateCredentialResolver;
exports.updateCurrentUser = require_users.updateCurrentUser;
exports.updateCurrentUserPassword = require_users.updateCurrentUserPassword;
exports.updateCurrentUserSettings = require_users.updateCurrentUserSettings;
exports.updateEnforceMfa = require_mfa.updateEnforceMfa;
exports.updateGlobalRole = require_users.updateGlobalRole;
exports.updateLdapConfig = require_ldap.updateLdapConfig;
exports.updateNpsSurveyState = require_npsSurvey.updateNpsSurveyState;
exports.updateOtherUserSettings = require_users.updateOtherUserSettings;
exports.updatePackage = require_communityNodes.updatePackage;
exports.updateProjectRole = require_roles.updateProjectRole;
exports.updateProvider = require_externalSecrets_ee.updateProvider;
exports.validatePasswordToken = require_users.validatePasswordToken;
exports.validateSignupToken = require_users.validateSignupToken;
exports.verifyMfaCode = require_mfa.verifyMfaCode;
