// @n8n/rest-api-client/dist/api/nodeTypes.cjs
require('../utils2.cjs');
const require_nodeTypes = require('../nodeTypes.cjs');

exports.fetchCommunityNodeAttributes = require_nodeTypes.fetchCommunityNodeAttributes;
exports.fetchCommunityNodeTypes = require_nodeTypes.fetchCommunityNodeTypes;
exports.getLocalResourceMapperFields = require_nodeTypes.getLocalResourceMapperFields;
exports.getNodeParameterActionResult = require_nodeTypes.getNodeParameterActionResult;
exports.getNodeParameterOptions = require_nodeTypes.getNodeParameterOptions;
exports.getNodeTranslationHeaders = require_nodeTypes.getNodeTranslationHeaders;
exports.getNodeTypeVersions = require_nodeTypes.getNodeTypeVersions;
exports.getNodeTypes = require_nodeTypes.getNodeTypes;
exports.getNodeTypesByIdentifier = require_nodeTypes.getNodeTypesByIdentifier;
exports.getNodesInformation = require_nodeTypes.getNodesInformation;
exports.getResourceLocatorResults = require_nodeTypes.getResourceLocatorResults;
exports.getResourceMapperFields = require_nodeTypes.getResourceMapperFields;
