// @n8n/rest-api-client/dist/api/users.cjs
require('../utils2.cjs');
const require_users = require('../users.cjs');

exports.changePassword = require_users.changePassword;
exports.deleteUser = require_users.deleteUser;
exports.generateInviteLink = require_users.generateInviteLink;
exports.getInviteLink = require_users.getInviteLink;
exports.getPasswordResetLink = require_users.getPasswordResetLink;
exports.getUsers = require_users.getUsers;
exports.login = require_users.login;
exports.loginCurrentUser = require_users.loginCurrentUser;
exports.logout = require_users.logout;
exports.sendForgotPasswordEmail = require_users.sendForgotPasswordEmail;
exports.setupOwner = require_users.setupOwner;
exports.signup = require_users.signup;
exports.submitPersonalizationSurvey = require_users.submitPersonalizationSurvey;
exports.updateCurrentUser = require_users.updateCurrentUser;
exports.updateCurrentUserPassword = require_users.updateCurrentUserPassword;
exports.updateCurrentUserSettings = require_users.updateCurrentUserSettings;
exports.updateGlobalRole = require_users.updateGlobalRole;
exports.updateOtherUserSettings = require_users.updateOtherUserSettings;
exports.validatePasswordToken = require_users.validatePasswordToken;
exports.validateSignupToken = require_users.validateSignupToken;
